misk-clustering/src/main/kotlin/misk/clustering/DefaultCluster.kt

```diff
@@ -74,6 +74,7 @@ class DefaultCluster(
   }
 
   /** Triggers a change to the cluster in response members becoming ready or not ready */
+  @JvmOverloads
```

Comment:
I've added `@JvmOverloads` in a few missing places, but not everywhere yet.
Undecided whether I should do that everywhere, or add a baseline file for detekt to ignore the existing violations (leaning towards the former).